### DISCUSSION TITLE
UEFI sanboot of RedHat fails due to SCSI error where transfer length is too large

### DISCUSSION THREAD

**Discussion by JPSmeets** [ipxe/ipxe#1071] [Q&A]
We have been using iPXE to sanboot RedHat in UEFI mode with our current SAN for a couple of years without problems.
While testing a new SAN however booting fails in shim with a "Unexpected return from initial read: Protocol error"
https://github.com/rhboot/shim/blob/main/shim.c line 942.
A tcpdump of the iSCSI traffic show that a READ10 failed as the transfer length of this READ10 exceeds the maximum transfer length.
I ran an INQUIRY to both the current and the new SAN and indeed the new SAN reports a much smaller maximum transfer length in the VPD than the current SAN.

Questions:
- Q1: normally the initiator is supposed to do an INQUIRY to determine the maximum transfer length. In iPXE I cannot find such an INQUIRY. Does iPXE not use an INQUIRY to determine the maximum transfer length (which would constitute a bug) ?
- Q2: How does iPXE determine the maximum transfer length (hardcoded) ? 

Note that these are SCSI, not iSCSI limits

**Comment by stappersg** [comment]
Questions:
* Q10: Why is this an iPXE issue?
* Q11: What is the maximum transfer length of the existing SAN?
* Q12: What is the maximum transfer length of the SAN in test?
* Q13: Can the maximum transfer length of the SAN in test be tweaked?
* Q14: What is the [MTU](https://en.wikipedia.org/wiki/Maximum_transmission_unit)  for each involved device?
* Q15: Which devices are involved?
